### DISCUSSION TITLE
title! and subtitle! flags

### DISCUSSION THREAD

**Discussion by ipozow** [Zrips/CMI#4499] [Enhancement]
**Description of feature/improvement request:**
It would be great to have the (-in:[ticks]) (-out:[ticks]) (-keep:[ticks]) flags in the specialized commands for title! and subtitle! like in the /cmi titlemsg command. 
---
### Basic explanation as to why this feature is needed and/or useful to have: ###
Useful so players can reach to read all the message before it disappears and still use the full potential of specialized commands.

**Comment by CloudeLecaw** [comment]
and ita not possible to use the titlecommand in this case? well still a nice to have idea

**Comment by ipozow** [comment]
Is not possible, I wanted to use it in the schedules so I can show it to everyone that meets the requirements and I can't do that with the titlemsg command. It would be perm:some.perm! cmi titlemsg [playerName] <message> and the flags, but playerName isn't a possible variable in the schedulers and allPlayers won't do the trick since it would send the message to everyone if only one online player has that perm

**Comment by CloudeLecaw** [comment]
Finaly your sheduler will not work
1. its nto possible to setup a sheduler that is just executed for some player
(Well it is but only if you use check for a single playername)

see here:

> If command starts with title! then all players will get title message defined after this variable
> If command starts with subtitle! then all players will get subtitle message defined after this variable

So i case of perm:some.perm! title! Hello My Friend
Everone will get it because the console that is allways executer of the scheduler has the permisison some.perm

**Comment by ipozow** [comment]
Is working fine on my server.   
It looks like this: - perm:anuncios.grupo! title! {#Gold}message
if I don't have that permission it doesn't show to me

**Comment by CloudeLecaw** [comment]
hm well good that it works even if it ahould not, so just lets hope it does. not het fixed one day :D
why? - Scheduler is allways a console meassage and if perm: is used then it should ask the schedule executer (the console - do you have thia perm?) and console has it. well thia does not count if you execute the scheduler from the ingame command for Tests. a scheduler is an automated process that has all rights and cant be interupted by a player. 

well still interesting to see this solution worked for you.. but since this is not the description how it should work i still would not recomment it. 
perm:pernission! usually is made for a command that gets executed By the player

**Comment by ipozow** [comment]
It doesn't work like that. Look at the config: “#   If command starts with perm:[permissionNode]! then command will get performed **if player** has permission node. This will not work when we dont know who is target player. In example: perm:cmi.announce.vip! broadcast! Hello vip guys”

**Comment by CloudeLecaw** [comment]
This is what i was saying the whole time *lol*  A Scheduler is executed from console and this one has all rights and that is why the command will be executed
on title! and! subtitle! you also dont know who is the target... 

create this test
```
BroadcastVoteReminderTest:
  Enabled: true
  MinPlayers: 1
  Delay: 5
  Repeat: true
  Commands:
  - perm:cmi.announce.vip! broadcast! Alle die das hier sehen, haben die richtige Berechtigung.
```

The Text sais: Everone that can see this, had the right permission

now i prodcasted to normal player + me and owner the sheduler

Result
![Snag_370c25d](https://user-images.githubusercontent.com/12080017/97768361-c8678e80-1b22-11eb-8c9d-2236fa3e87ea.png)

The owner and i have the permission
The player does nothave the permission

the is prove that i was right
the perm asked for is the executer and this is the Scheduler that is allways console with all permission

**Comment by ipozow** [comment]
“This will not work when we don't know who is target player”. With title! and subtitle! we know who is target player. With broadcast! we don't.
